src/scipp/data/__init__.py

```diff
@@ -69,16 +69,22 @@ def vulcan_steel_strain_data() -> DataArray:
     return load_hdf5(get_path('VULCAN_221040_processed.h5'))
 
 
-def table_xyz(nrow: int) -> DataArray:
+def table_xyz(nrow: int, coord_max=1, coord_dtype=None) -> DataArray:
     """
     Return a 1-D data array ("table") with x, y, and z coord columns.
     """
     from numpy.random import default_rng
 
     rng = default_rng(seed=1234)
-    x = array(dims=['row'], unit='m', values=rng.random(nrow))
-    y = array(dims=['row'], unit='m', values=rng.random(nrow))
-    z = array(dims=['row'], unit='m', values=rng.random(nrow))
+    x = array(
+        dims=['row'], unit='m', values=rng.random(nrow) * coord_max, dtype=coord_dtype
+    )
+    y = array(
+        dims=['row'], unit='m', values=rng.random(nrow) * coord_max, dtype=coord_dtype
+    )
+    z = array(
+        dims=['row'], unit='m', values=rng.random(nrow) * coord_max, dtype=coord_dtype
+    )
```

Comment:
Not sure it matters much in practice, but now this function will be slower, even with the default `coord_max`. Unless you think this is more generally useful, maybe you can keep the scaling in the unit test where it is required?

Reply:
I think it might be useful when testing with integer coordinates.
But I'll change it so it doesn't increase the runtime when used without scaling.